Frontend/business-expense-manager/src/components/card/card.component.ts

```diff
@@ -0,0 +1,58 @@
+import { NgFor, NgIf } from '@angular/common';
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Card } from '../../models/card.model';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+
+@Component({
+  selector: 'app-card',
+  standalone: true,
+  imports: [NgFor, NgIf, MatProgressSpinnerModule],
+  templateUrl: './card.component.html',
+  styleUrl: './card.component.css',
+})
+export class CardComponent {
+
+  @Input() card: Card = {
+    title: 'Default',
+    balanceAmount: 5400,
+    goalAmount: 7700,
+    type: 'Goal',
+    colour: 'Yellow'
+  };
+
+  @Output() cardClicked = new EventEmitter<Card>();
+
+  constructor() {}
+
+  get progressAmount() : number {
+    return (this.card.balanceAmount / this.card.goalAmount) * 100;
+  }
+
+  get progressAmountString() : string {
+    let temp = (this.card.balanceAmount / this.card.goalAmount) * 100;
+    if(temp === 100) {
+      return "100.0";
+    }
+    return temp.toFixed(2);
+  }
+
+  get statusColour(): string {
```

Comment:
hmmm, what happens if I change the card's color manually the the dev tool? Will it change the status?

Reply:
Interesting point - I don't think you can change the object itself that is passed in, I don't look at it's actual colour like on the page. I figured the colour passed in by the parent would be the best way to deliver it's status as well, instead of making another variable in the object.